src/elements/content-sidebar/Sidebar.js

```diff
@@ -96,6 +101,14 @@ class Sidebar extends React.Component<Props, State> {
         this.setForcedByLocation();
     }
 
+    componentDidMount() {
+        const { file, api, metadataSidebarProps, docGenSidebarProps }: Props = this.props;
+        // if docgen feature is enabled, load metadata to check whether file is a docgen template
+        if (docGenSidebarProps.enabled) {
+            docGenSidebarProps.checkDocGenTemplate(api, file, metadataSidebarProps.isFeatureEnabled);
+        }
```

Comment:
I didn't use the logic in ContentSidebar because it only fetches metadata in following case:
`        const canHaveMetadataSidebar = !isFeatureEnabled && SidebarUtils.canHaveMetadataSidebar(this.props);
`
which is confusing to me since it seems to be the opposite of metadata being enabled. So I didn't want to change the existing logic that would affect metadata tab (since I don't understand it) and created my own.

Reply:
but if it is indeed a mistake then I can try updating that indeed - I raised it with the metadata team

Reply:
I discussed it with @bfoxx1906  and decided t o leave it as it is - the code in the current implementation will be removed once we retrieve information of whether a file is a DocGen template from backend - the logic in ContentSidebar is quite old and affecting state of that component and Metadata sidebar so I'd like to avoid modifying it since I'll have to revert back once the logic is no longer necessary.